celery/concurrency/thread.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+"""Thread execution pool."""
+
+from concurrent.futures import wait, ThreadPoolExecutor
+from .base import BasePool, apply_target
+
+__all__ = ('TaskPool',)
+
+class ApplyResult(object):
+    def __init__(self, future):
+        self.f = future
+        self.get = self.f.result
+
+    def wait(self, timeout=None):
+        wait([self.f], timeout)
+
+class TaskPool(BasePool):
+    """Thread Task Pool."""
+
+    body_can_be_buffer = True
+    signal_safe = False
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```

Comment:
`super()` calls are only valid with Python 3.
Celery 4.x still supports Python 2.7 so we'll need to adjust that.